src/editor/overlay_widget.cpp

```diff
@@ -447,62 +482,95 @@ EditorOverlayWidget::fill()
+    input_tile(pos, tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y)), false);
 
     Vector pos_(0.0f, 0.0f);
 
     // Going left...
     pos_ = pos + Vector(-1, 0);
-    if (pos_.x >= 0) {
+    if (pos_.x >= 0)
+    {
       if (check_tiles_for_fill(replace_tile,
           tilemap->get_tile_id(static_cast<int>(pos_.x), static_cast<int>(pos_.y)),
-          tiles->pos(static_cast<int>(tpos.x - 1), static_cast<int>(tpos.y)))) {
+          tiles->pos(static_cast<int>(tpos.x - 1), static_cast<int>(tpos.y))))
+      {
         pos_stack.push_back( pos_ );
         continue;
       }
     }
 
     // Going right...
     pos_ = pos + Vector(1, 0);
-    if (pos_.x < static_cast<float>(tilemap->get_width())) {
+    if (pos_.x < static_cast<float>(tilemap->get_width()))
+    {
       if (check_tiles_for_fill(replace_tile,
           tilemap->get_tile_id(static_cast<int>(pos_.x), static_cast<int>(pos_.y)),
-          tiles->pos(static_cast<int>(tpos.x + 1), static_cast<int>(tpos.y)))) {
+          tiles->pos(static_cast<int>(tpos.x + 1), static_cast<int>(tpos.y))))
+      {
         pos_stack.push_back( pos_ );
         continue;
       }
     }
 
     // Going up...
     pos_ = pos + Vector(0, -1);
-    if (pos_.y >= 0) {
+    if (pos_.y >= 0)
+    {
       if (check_tiles_for_fill(replace_tile,
           tilemap->get_tile_id(static_cast<int>(pos_.x), static_cast<int>(pos_.y)),
-          tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y - 1)))) {
+          tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y - 1))))
+      {
         pos_stack.push_back( pos_ );
         continue;
       }
     }
 
     // Going down...
     pos_ = pos + Vector(0, 1);
-    if (pos_.y < static_cast<float>(tilemap->get_height())) {
+    if (pos_.y < static_cast<float>(tilemap->get_height()))
+    {
       if (check_tiles_for_fill(replace_tile,
           tilemap->get_tile_id(static_cast<int>(pos_.x), static_cast<int>(pos_.y)),
-          tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y + 1)))) {
+          tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y + 1))))
+      {
         pos_stack.push_back( pos_ );
         continue;
       }
     }
 
     // Autotile happens after directional detection (because of borders; see snow tileset)
-    if (g_config->editor_autotile_mode) {
-      input_autotile(pos, tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y)));
-    }
+    if (g_config->editor_autotile_mode)
+      input_autotile(pos, tiles->pos(static_cast<int>(tpos.x), static_cast<int>(tpos.y)), false);
 
     // When tiles on each side are already filled or occupied by another tiles, it ends.
     pos_stack.pop_back();
   }
+  tilemap->check_state();
+}
+
+void
+EditorOverlayWidget::replace()
+{
+  auto tilemap = m_editor.get_selected_tilemap();
+  Uint32 replace_tile = tilemap->get_tile_id(static_cast<int>(m_hovered_tile.x), static_cast<int>(m_hovered_tile.y));
```

Comment:
Please note that `Uint32` is not the same thing as `uint32_t`

Reply:
I used the same data type as is used in the fill() function. I assume it is used for a reason. If not, I can change it.

Reply:
Alright